tests/models/chinese_clip/test_processor_chinese_clip.py

```diff
@@ -73,26 +74,30 @@ def setUp(self):
             "image_std": [0.26862954, 0.26130258, 0.27577711],
             "do_convert_rgb": True,
         }
-        self.image_processor_file = os.path.join(self.tmpdirname, FEATURE_EXTRACTOR_NAME)
-        with open(self.image_processor_file, "w", encoding="utf-8") as fp:
+        cls.image_processor_file = os.path.join(cls.tmpdirname, FEATURE_EXTRACTOR_NAME)
+        with open(cls.image_processor_file, "w", encoding="utf-8") as fp:
             json.dump(image_processor_map, fp)
 
-        tokenizer = self.get_tokenizer()
-        image_processor = self.get_image_processor()
+        tokenizer = cls.get_tokenizer()
+        image_processor = cls.get_image_processor()
         processor = ChineseCLIPProcessor(tokenizer=tokenizer, image_processor=image_processor)
-        processor.save_pretrained(self.tmpdirname)
+        processor.save_pretrained(cls.tmpdirname)
```

Comment:
If you expand this file, you will see we have tests like `test_save_load_pretrained_default` and it contains

> processor_fast.save_pretrained(self.tmpdirname)

this means `cls.tmpdirname` is (potentially) modified by (at least one) test.

Previously while we use `setup`, it's unlikely to cause problem. While now we use `setupClass`, it's no longer clear because multiple also want to read the same `cls.tmpdirname`.

 We will have to do something like in `tests/test_processing_common.py::ProcessorTesterMixin::test_processor_from_and_save_pretrained`
 
 ```
        with tempfile.TemporaryDirectory() as tmpdirname:
            saved_files = processor_first.save_pretrained(tmpdirname)
```

I don't know which other tests have to be checked however.

Reply:
I checked all of the affected files :sweat_drops:

Whenever tests write to `cls.tmpdirname` I replaced that with a new `tmpdir`, so the test remains independent.